digicampipe/io/zfits.py

```diff
@@ -47,11 +47,11 @@ def zfits_event_source(
 
     loaded_telescopes = []
 
-    for event in protozfitsreader.ZFile(url):
-        if max_events is not None and event.event_id > max_events:
+    for event_counter, event in enumerate(protozfitsreader.ZFile(url)):
+        if max_events is not None and event_counter > max_events:
             break
 
-        data.r0.event_id = event.event_id
+        data.r0.event_id = event_counter
```

Comment:
`data.r0.event_id = event.event_number` ?

Reply:
???

Reply:
`event_counter` is taken from the `enumerate()` so we don't need to store it as we said.
If I look at the `Event` class then the only attribute I find close to a unique identifier is `event.event_number`

Reply:
Yes, but see below: `event.event_number` is already assigned to `data.r0.tel[tel_id].camera_event_number`.
We do not want to duplicate this do we?

Reply:
ok did not see it... sorry.
So I think we should get rid of those counters in the future. Naming should be : event_id for unique identification number and event_counter if needed somewhere.

Reply:
Actually this is totally messed up:
```python
class R0Container(Container):
    event_id = Field(-1, "event id number")
    tel = Field(Map(R0CameraContainer), "map of tel_id to R0CameraContainer")

class R0CameraContainer(Container):
    camera_event_id = Field(int, 'Camera event number')
    camera_event_number = Field(int, "camera event number")
```

3 event ids ... or numbers ...

Reply:
I think it is supposed to be like this

```python
class R0Container(Container):
    array_event_id = Field(-1, "Array event identification number")
    tel = Field(Map(R0CameraContainer), "map of tel_id to R0CameraContainer")

class R0CameraContainer(Container):
    camera_event_id = Field(int, 'Camera event identification number')
```

Reply:
or maybe using the same naming for both array and telescope level is better. 

that way you can do : 

`data.r0.event_id`

and

`data.r0.tel[x].event_id`

Reply:
Looked at the code, indeed the random access was not interfaced to python. So it should only be readEvent() and rewindTable(). 

I also fixed the crash after reading to many events, but I am not sure what should be the behavior there:
- looks like I can raise a python exception from C++
- or I can just return an integer object instead of a protobuf object

Reply:
Thanks!

Where is the random access missing actually? Are we supposed to call a function existing in `rawzfitsreader` or do you miss it there?